packages/nx/src/migrations/update-17-0-0/move-cache-directory.ts

```diff
@@ -0,0 +1,16 @@
+import { Tree } from '../../generators/tree';
+import ignore from 'ignore';
+
+export default function moveCacheDirectory(tree: Tree) {
+  const gitignore = tree.exists('.gitignore')
+    ? tree.read('.gitignore', 'utf-8')
+    : '';
+  const ig = ignore();
+  ig.add(gitignore);
+  if (!ig.ignores('.nx/cache')) {
```

Comment:
Thanks for keeping this specific, we will have a need for some tracked files in `.nx` soon!

Reply:
The wrapper also lives here and needs to be checked in so it had to be specific 😃